flex/engines/graph_db/database/manager.h

```diff
@@ -0,0 +1,96 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ENGINES_GRAPH_DB_APP_MANAGER_H_
+#define ENGINES_GRAPH_DB_APP_MANAGER_H_
+
+#include <nlohmann/json.hpp>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "flex/engines/graph_db/database/graph_db.h"
+#include "flex/engines/graph_db/database/graph_db_session.h"
+#include "flex/engines/graph_db/database/manager.h"
+#include "flex/utils/service_utils.h"
+
+namespace gs {
+
+struct VertexData {
+  gs::Any pk_value;
+  gs::label_t label_id;
+  std::vector<std::pair<std::string, gs::Any>> properties;
+  VertexData() {}
+  ~VertexData() {}
+};
+
+struct EdgeData {
+  gs::label_t src_label_id, dst_label_id, edge_label_id;
+  gs::Any src_pk_value, dst_pk_value;
+  gs::Any property_value;
+  std::string property_name;
+  EdgeData() {}
+  ~EdgeData() {}
+};
+
+gs::VertexData inputVertex(const nlohmann::json& vertex_json,
+                           const nlohmann::json& schema_json, int shard_id);
+
+gs::EdgeData inputEdge(const nlohmann::json& edge_json,
+                       const nlohmann::json& schema_json, int shard_id);
+
+class VertexEdgeManager {
+ public:
+  // check schema
+  static void checkVertexSchema(const nlohmann::json& schema_json,
```

Comment:
Should we return a boolean for the chosen functions?

Reply:
I've modified the return values of some functions to result/bool/status to ensure that detailed error causes are returned in case of error, while avoiding too many layers of try+throw